src/lib/Imp.hs

```diff
@@ -73,22 +75,27 @@ data ImpCatEnv = ImpCatEnv
 type ImpM = ExceptT () (ReaderT ImpCtx (Cat ImpCatEnv))
 type AtomRecon = Abs (Nest Binder) Atom
 
-toImpModule :: Backend -> CallingConvention -> Name
+toImpModule :: TopEnv -> Backend -> CallingConvention -> Name
             -> [IBinder] -> Maybe Dest -> Block
             -> (ImpFunction, ImpModule, AtomRecon)
-toImpModule backend cc entryName argBinders maybeDest block =
-  runImpM initOpts inVarScope $ do
+toImpModule env backend cc entryName argBinders maybeDest block = do
```

Comment:
We should really rewrite this function to take a whole bunch of blocks or function atoms and produce an Imp module that contains all of them

Reply:
Yes, that might be a good way to factor it.